packages/sdk/contractkit/src/kit.ts

```diff
@@ -252,242 +208,135 @@ export class ContractKit {
    * @param token cUSD (StableToken) or CELO (GoldToken)
    */
   async setFeeCurrency(token: CeloToken): Promise<void> {
-    this.config.feeCurrency =
+    const address =
       token === CeloContract.GoldToken ? undefined : await this.registry.addressFor(token)
+    if (address) {
+      await this.updateGasPriceInConnectionLayer(address)
+    }
+    this.connection.defaultFeeCurrency = address
+  }
+
+  // TODO: remove once cUSD gasPrice is available on minimumClientVersion node rpc
+  async updateGasPriceInConnectionLayer(currency: Address) {
+    const gasPriceMinimum = await this.contracts.getGasPriceMinimum()
+    const rawGasPrice = await gasPriceMinimum.getGasPriceMinimum(currency)
+    const gasPrice = rawGasPrice.multipliedBy(this.gasPriceSuggestionMultiplier).toFixed()
+    await this.connection.setGasPriceForCurrency(currency, gasPrice)
+  }
+
+  async getEpochSize(): Promise<number> {
+    const validators = await this.contracts.getValidators()
+    const epochSize = await validators.getEpochSize()
+
+    return epochSize.toNumber()
+  }
+
+  async getFirstBlockNumberForEpoch(epochNumber: number): Promise<number> {
+    const epochSize = await this.getEpochSize()
+    // Follows GetEpochFirstBlockNumber from celo-blockchain/blob/master/consensus/istanbul/utils.go
+    if (epochNumber === 0) {
+      // No first block for epoch 0
+      return 0
+    }
+    return (epochNumber - 1) * epochSize + 1
   }
 
+  async getLastBlockNumberForEpoch(epochNumber: number): Promise<number> {
+    const epochSize = await this.getEpochSize()
+    // Follows GetEpochLastBlockNumber from celo-blockchain/blob/master/consensus/istanbul/utils.go
+    if (epochNumber === 0) {
+      return 0
+    }
+    const firstBlockNumberForEpoch = await this.getFirstBlockNumberForEpoch(epochNumber)
+    return firstBlockNumberForEpoch + (epochSize - 1)
+  }
+
+  async getEpochNumberOfBlock(blockNumber: number): Promise<number> {
+    const epochSize = await this.getEpochSize()
+    // Follows GetEpochNumber from celo-blockchain/blob/master/consensus/istanbul/utils.go
+    const epochNumber = Math.floor(blockNumber / epochSize)
+    if (blockNumber % epochSize === 0) {
+      return epochNumber
+    } else {
+      return epochNumber + 1
+    }
+  }
+
+  // *** NOTICE ***
+  // Next functions exists for backwards compatibility
+  // These should be consumed via connection to avoid future deprecation issues
+
   addAccount(privateKey: string) {
-    assertIsCeloProvider(this.web3.currentProvider)
-    this.web3.currentProvider.addAccount(privateKey)
+    this.connection.addAccount(privateKey)
```

Comment:
are these exposed on `contractkit` for backwards compatibility?
any reason why `kit.connection` isn't sufficient

Reply:
maybe we need a depreacted decorator, that logs a warning first time we use this method